sympy/polys/rings.py

```
@@ -3094,3 +3446,787 @@ def sqf_list(f, all=False):
+
+    def to_dense(self):
+        return NotImplementedError
+
+    def to_dict(self):
+        return self.flint_poly.to_dict()
+
+    @property
+    def is_generator(self):
+        return self in self.ring._gens_set
+
+    @property
+    def is_ground(self):
+        if self.flint_poly.is_zero():
+            return True
+        terms = self.flint_poly.to_dict()
+        return len(terms) == 1 and self.ring.zero_monom in terms
+
+    @property
+    def is_monomial(self):
+        return self.flint_poly.is_zero() or (len(self.flint_poly.to_dict()) == 1 and self.LC == 1)
+
+    @property
+    def is_term(self):
+        return len(self.flint_poly.to_dict()) <= 1
+
+    @property
+    def is_negative(self):
+        return self.ring.domain.is_negative(self.LC)
+
+    @property
+    def is_positive(self):
+        return self.ring.domain.is_positive(self.LC)
+
+    @property
+    def is_nonnegative(self):
+        return self.ring.domain.is_nonnegative(self.LC)
+
+    @property
+    def is_nonpositive(self):
+        return self.ring.domain.is_nonpositive(self.LC)
+
+    @property
+    def is_zero(self):
+        return self.flint_poly.is_zero()
+
+    @property
+    def is_one(self):
+        return self == self.ring.one
+
+    @property
+    def is_monic(self):
+        return self.ring.domain.is_one(self.LC)
+
+    @property
+    def is_primitive(self):
+        content, _ = self.flint_poly.factor()
+        return self.ring.domain.is_one(content)
+
+    @property
+    def is_linear(self):
+        return all(sum(monom) <= 1 for monom in self.itermonoms())
+
+    @property
+    def is_quadratic(self):
+        return all(sum(monom) <= 2 for monom in self.itermonoms())
+
+    @property
+    def is_squarefree(self):
+        if not self.ring.ngens:
+            return True
+        _, factors = self.flint_poly.factor_squarefree()
+        return all(exp == 1 for _, exp in factors)
+
+    @property
+    def is_irreducible(self):
+        return NotImplementedError
```

NotImplementedError should be raised rather than returned.